DOCS-3910 Advanced Options for Browser Test Steps Edits (#16369)
DataDog/documentation: content/en/synthetics/browser_tests/advanced_options.md

  - /synthetics/guide/browser-tests-switch-tabs/
  - /synthetics/guide/browser-test-self-maintenance/
further_reading:
- link: "https://www.datadoghq.com/blog/browser-tests/"
  tag: "Blog"
  text: "User experience monitoring with Synthetic browser tests"
- link: "/synthetics/browser_tests/actions/"
  tag: "Documentation"
  text: "Learn more about Browser Test Steps"
---

## Overview

> This page describes how Synthetic browser tests are self maintained and the various advanced options available in the recording workflow. 

Reviewer: is it a good practice to have the 1st sentence of a page saying "This page describes ..."

Author: Yep, it's a common practice across our doc set to basically introduce the topics in this doc page.

Author: How about something like this?

```suggestion
This page describes how Synthetic browser tests locate elements with the Datadog algorithm and the Datadog user locator system. 
```

Reviewer: yup and if you think that doesn't bring any value we should probably drop it

## Test maintenance

Flakiness is a pain point in end-to-end testing because tests occasionally fail when a frontend team implements changes, causing an identifier in your test to alert instead of an actual application issue.

To prevent flaky tests, Datadog uses an algorithm that leverages a set of locators to target elements in browser tests. A small change in the UI may modify an element (for example, moving it to another location). The browser test automatically locates the element again based on points of reference that were not affected by the change. 

When the test runs successfully, the browser test recomputes (or "self heals") any broken locators with updated values, ensuring your tests do not break from simple UI updates and that your tests are automatically adapting to your application's UI. 

To ensure that your browser test does not validate an unexpected change, use [assertions][5] in your test creation. Assertions allow you to define what is and what is not expected behavior associated with the test step journey. 

> ## User specified locator

Author: ```suggestion
### User specified locator
```

  - /synthetics/guide/browser-tests-switch-tabs/
  - /synthetics/guide/browser-test-self-maintenance/
further_reading:
- link: "https://www.datadoghq.com/blog/browser-tests/"
  tag: "Blog"
  text: "User experience monitoring with Synthetic browser tests"
- link: "/synthetics/browser_tests/actions/"
  tag: "Documentation"
  text: "Learn more about Browser Test Steps"
---

## Overview

This page describes how Synthetic browser tests are self maintained and the various advanced options available in the recording workflow. 

## Locate an element

> ## Datadog algorithm

Reviewer: ```suggestion
### Datadog algorithm
```
